src/sentry/buffer/redis.py

```diff
@@ -233,19 +242,32 @@ def process_pending(self, partition=None):
 
         try:
             keycount = 0
-            with self.cluster.all() as conn:
-                results = conn.zrange(pending_key, 0, -1)
-
-            with self.cluster.all() as conn:
-                for host_id, keys in results.value.items():
-                    if not keys:
-                        continue
-                    keycount += len(keys)
-                    for key in keys:
-                        pending_buffer.append(key.decode("utf-8"))
-                        if pending_buffer.full():
-                            process_incr.apply_async(kwargs={"batch_keys": pending_buffer.flush()})
-                    conn.target([host_id]).zrem(pending_key, *keys)
+            if self.is_redis_cluster:
+                keys = self.cluster.zrange(pending_key, 0, -1)
```

Comment:
Does this imply that the pending key will now be located on only one node, vs being on multiple nodes in the previous implementation? Are there any concerns around scale there?

Reply:
yes, i would increase `pending_partitions` to compensate for that